cmd/proxy/main.go

```diff
@@ -329,28 +329,43 @@ func (p *Proxy) runUDSMasterServer(ctx context.Context, o *ProxyRunOptions, serv
 	return stop, nil
 }
 
-func (p *Proxy) runMTLSMasterServer(ctx context.Context, o *ProxyRunOptions, server *agentserver.ProxyServer) (StopFunc, error) {
-	var stop StopFunc
-
-	proxyCert, err := tls.LoadX509KeyPair(o.serverCert, o.serverKey)
+func (p *Proxy) getTLSConfig(caFile, certFile, keyFile string) (*tls.Config, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		return nil, fmt.Errorf("failed to load X509 key pair %s and %s: %v", o.serverCert, o.serverKey, err)
+		return nil, fmt.Errorf("failed to load X509 key pair %s and %s: %v", certFile, keyFile, err)
 	}
+
+	if caFile == "" {
+		return &tls.Config{Certificates: []tls.Certificate{cert}}, nil
+	}
+
 	certPool := x509.NewCertPool()
-	caCert, err := ioutil.ReadFile(o.serverCaCert)
+	caCert, err := ioutil.ReadFile(caFile)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read server CA cert %s: %v", o.serverCaCert, err)
+		return nil, fmt.Errorf("failed to read cluster CA cert %s: %v", caFile, err)
```

Comment:
s/cluster//

This can be the cluster CA or the server CA.

Reply:
done